mongoose.c

```diff
@@ -4909,13 +4912,33 @@ static void read_conn(struct mg_connection *c, struct pkt *pkt) {
   }
 }
 
+// process options (MSS)
+static void handle_opt(struct connstate *s, struct tcp *tcp) {
+  uint8_t *opts = (uint8_t *) (tcp + 1);
+  int len = 4 * ((int) (tcp->off >> 4) - (sizeof(*tcp) / 4));
+  s->dmss = 536;     // assume default, RFC-9293 3.7.1
+  while (len > 0) {  // RFC-9293 3.1 3.2
+    uint8_t kind = opts[0], optlen = 1;
+    if (kind != 1) {         // No-Operation
+      if (kind == 0) break;  // End of Option List
+      optlen = opts[1];
+      if (kind == 2 && optlen == 4)  // set received MSS
+        s->dmss = (uint16_t) (((uint16_t) opts[2] << 8) + opts[3]);
+    }
+    MG_INFO(("kind: %u, optlen: %u, len: %d\n", kind, optlen, len));
```

Comment:
MG_DEBUG !

Reply:
Ouch, leftover, that one intended to be a _VERBOSE... sorry about that.